iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -600,7 +600,10 @@ private ColumnStatistics readColStats(Table table, Path statsPath) {
 
   @Override
   public boolean canComputeQueryUsingStats(org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
-    if (getStatsSource().equals(HiveMetaHook.ICEBERG) && hmsTable.getMetaTable() == null) {
+    if (hmsTable.getMetaTable() != null) {
```

Comment:
Query against Iceberg Branch/Tag can also benefit from the stats. We can optimize this later.